### PR TITLE
snap-bootstrap: set expected filesystem labels

### PR DESCRIPTION
**Pull request by cmatsuoka**
When creating filesystems that expect specific labels, set them based
on the structure role.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

### CODE REVIEW

**Comment by mvo5** [comment]
I wonder how hard it would be to add a check for this to the uc20-snap-recovery test?

File: cmd/snap-bootstrap/partition/sfdisk.go [hunk 1]
```diff
@@ -222,7 +227,14 @@ func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume)
 		fmt.Fprintf(buf, "%s : start=%12d, size=%12d, type=%s, name=%q\n", node, p.StartOffset/sectorSize,
 			s.Size/sectorSize, partitionType(ptable.Label, p.Type), s.Name)
 
-		// Are roles unique so we can use it to map nodes? Should we use labels instead?
+		// Set expected labels based on role
+		switch s.Role {
+		case systemBootRole:
```
Comment (bboozzoo) [inline cmd/snap-bootstrap/partition/sfdisk.go]: We could use the role names defined in the gadget package. i.e.:
```
diff --git a/cmd/snap-bootstrap/partition/sfdisk.go b/cmd/snap-bootstrap/partition/sfdisk.go
index fd80b7b0ef..7347362905 100644
--- a/cmd/snap-bootstrap/partition/sfdisk.go
+++ b/cmd/snap-bootstrap/partition/sfdisk.go
@@ -31,8 +31,6 @@ import (
 )
 
 const (
-       systemBootRole  = "system-boot"
-       systemDataRole  = "system-data"
        ubuntuBootLabel = "ubuntu-boot"
        ubuntuDataLabel = "ubuntu-data"
 
@@ -229,9 +227,9 @@ func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume)
 
                // Set expected labels based on role
                switch s.Role {
-               case systemBootRole:
+               case gadget.SystemBoot:
                        s.Label = ubuntuBootLabel
-               case systemDataRole:
+               case gadget.SystemData:
                        s.Label = ubuntuDataLabel
                }
 
```

Comment (cmatsuoka) [reply cmd/snap-bootstrap/partition/sfdisk.go]: Using the names already defined is much better, thanks.

**Comment by cmatsuoka** [comment]
> I wonder how hard it would be to add a check for this to the uc20-snap-recovery test?

I'll add something there.

**Comment by bboozzoo** [comment]
Some conflicts in tests/main/uc20-snap-recovery/task.yaml

**Comment by cmatsuoka** [comment]
Force-pushed to fix the merge commit author (tried to use the web editor to solve the conflict) and not fail the CLA.

**Pull request merged**